lib/friendly_shipping/services/usps_ship/parse_rate_estimates_response.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module FriendlyShipping
+  module Services
+    class USPSShip
+      class ParseRateEstimatesResponse
+        extend Dry::Monads::Result::Mixin
+        CURRENCY = Money::Currency.new('USD').freeze
+
+        class << self
+          # Parse a rate estimates response.
+          #
+          # @param request [Request] the request that was used to obtain this response
+          # @param response [Response] the response that USPS returned
+          # @return [Result<ApiResult<Array<Rate>>>] the {Rate}s wrapped in an {ApiResult} object
+          def call(request:, response:)
+            rates = JSON.parse(response.body)['rates'].map do |rate|
```

Comment:
Does USPS always return rates? Do you need to protect against malformed responses (like UPS)?

Reply:
I haven't seen any malformed responses yet, but it's a good idea. I added some basic parser error handling.